packages/analytics-node/src/transports/http.ts

```diff
@@ -49,7 +49,7 @@ export class Http extends BaseTransport implements Transport {
           }
         });
       });
-      req.on('error', this.buildResponse.bind(this));
+      req.on('error', () => resolve(null));
```

Comment:
Correctly resolve promise on `error`

Reply:
@justin-fiedler 
1. An original network error is thrown away here - it could be logged for debugging.
2. Other transports do rejects (for example, https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-browser/src/transports/send-beacon.ts#L28) - errors are caught/logged by callers.